### PR TITLE
Add line simple object and selection for all

### PR DESCRIPTION
**Pull request by JerdoAve**
- change approach for saving of simple objects 
- implement highlighting hover and selection behaviour
- add new simple object (line)

### CODE REVIEW

**Comment by JerdoAve** [comment]
Hi Team,

I'm following your progress about Ketcher and I really like all the things around it.

I've decided to help a bit, and please have a look at my PR.

All the changes are included in the description.

Regards,
JA

**Comment by AndreiMazol** [comment]
Dear @JerdoAve ,

thank you for your help in making our product better.

We will review your PR asap.

Best regards,
Ketcher team

File: .vscode/launch.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
   "version": "0.2.0",
   "configurations": [
+      "type": "pwa-chrome",
```
Comment (AndreiMazol) [inline .vscode/launch.json]: Please fix merge conflict

File: src/script/format/chemGraph/toGraph/prepare.js [hunk 2]
```diff
@@ -40,17 +40,16 @@ export function prepareStructForGraph(struct) {
   const molFrags = Array.from(struct.frags.keys()).filter(
     fid => !rgFrags.has(fid)
   )
-  const moleculeFragsAtoms = molFrags.reduce((acc, fid) => {
-    const fragAtoms = struct.getFragmentIds(fid)
-    acc = acc.union(fragAtoms)
-    return acc
-  }, new Pile())
+  for (const fid of molFrags) {
```
Comment (AndreiMazol) [inline src/script/format/chemGraph/toGraph/prepare.js]: Please revert this changes: it is necessary to save structure as a single molecule

File: src/script/render/restruct/resimpleobject.js [hunk 3]
```diff
@@ -20,6 +20,10 @@ import draw from '../draw'
 import util from '../util'
 import scale from '../../util/scale'
 import Vec2 from 'src/script/util/vec2'
+import { maxHeaderSize } from 'http'
+import { ENGINE_METHOD_PKEY_ASN1_METHS } from 'constants'
```
Comment (AndreiMazol) [inline src/script/render/restruct/resimpleobject.js]: unused imports should be deleted

File: src/script/ui/app/toolbar.jsx [hunk 4]
```diff
@@ -134,9 +134,8 @@ const template = [
     id: 'template-common',
     component: TemplatesList
   },
-  'template-lib'
-  //TODO: it should be enabled after starting work on enhanced stereo
-  //'enhanced-stereo'
+  'template-lib',
+  'enhanced-stereo'
```
Comment (AndreiMazol) [inline src/script/ui/app/toolbar.jsx]: revert this please

File: src/script/util/scale.js [hunk 5]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  ***************************************************************************/
 
+import Vec2 from './vec2'
```
Comment (AndreiMazol) [inline src/script/util/scale.js]: please delete unused import

File: src/script/util/box2abs.js [hunk 6]
```diff
@@ -97,7 +97,10 @@ Box2Abs.prototype.translate = function (d) {
  */
 Box2Abs.prototype.transform = function (f, options) {
   console.assert(!!f)
-  return new Box2Abs(f(this.p0, options), f(this.p1, options))
+  return new Box2Abs(
+    f(this.p0 || this.pos[0], options),
+    f(this.p1 || this.pos[1], options)
```
Comment (AndreiMazol) [inline src/script/util/box2abs.js]: I guess the second part of condition has been added by mistake

**Review by AndreiMazol — CHANGES_REQUESTED** [review]

**Comment by AndreiMazol** [comment]
Dear @JerdoAve ,

I really  appreciate your hard work! All looks great! I see that some my changes were overwritten by mistake so I have reverted them. 

Looking forward for our further collaboration.

Best regards,
Andrei from Ketcher team

**Review by AndreiMazol — APPROVED** [review]
